src/core/src/types/arrays.rs

```diff
@@ -73,28 +59,13 @@ impl<T: Zero + Clone> Array2<T> {
     /// a.resize_if_different((8, 9));
     /// assert_eq!(a[(3, 3)], 0.0);
     /// ```
-    pub fn resize_if_different(&mut self, size: (Ix, Ix)) {
-        if self.0.shape() != &[size.0, size.1] {
+    pub fn resize_if_different(&mut self, size: (usize, usize)) {
```

Comment:
The thing with your solution is that it does not preserve the content of the array (because of the multi-dimensional aspect if the size of an axis that is not the zero axis changes, all the indexes will be offset).

In the cases where we use it, wouldn't it be simpler to just not use it ?
For example [here](https://github.com/lumol-org/lumol/blob/740eee0732bd424c79c7aa3dd9f1bc7bb9cb74bd/src/core/src/sys/cache.rs#L84) having instead the following seems OK:
``` rust
self.pairs_cache = Array2::zeros((system.size(), system.size()));
```

Or, if the time for initializing all the values becomes an issue, have an unsafe function create an uninitialized array, since in both use cases we do not care about the values inside the array and overwrite every element.